src/ReferenceList.php

```diff
@@ -194,15 +166,23 @@ public function hasReferenceHash( $referenceHash ) {
 	 * @param string $referenceHash	`
 	 */
 	public function removeReferenceHash( $referenceHash ) {
-		$reference = $this->getReference( $referenceHash );
-
-		if ( $reference !== null ) {
-			$this->detach( $reference );
+		foreach ( $this->references as $index => $reference ) {
+			if ( $reference->getHash() === $referenceHash ) {
+				$this->removeReferenceAtIndex( $index );
+			}
 		}
 	}
 
 	/**
-	 * Returns the reference with the provided hash, or null if there is no such reference in the list.
+	 * @param int $index
+	 */
+	private function removeReferenceAtIndex( $index ) {
```

Comment:
Typically I find single-line methods like this one confusing and not helpful. Did you planned to make this public?